src/components/page/WithGATracker/index.tsx

```diff
@@ -30,32 +43,47 @@ export const getGAusername = (): string => username;
  * @param {string} page the url string of the page view being tracked
  * @param {FlexObject} options tracking options for the page view
  */
-export const trackPage = (page: string, options: FlexObject = {}): void => {
-  GoogleAnalytics.set({
-    page,
-    ...options,
-  });
-  GoogleAnalytics.pageview(page);
+export const trackPage = (page: string, options?: FlexObject): void => {
+  const gaCode = getGaCode(options);
+  if (gaCode && gaCode.length) {
+    GoogleAnalytics.set({
+      page,
+      ...options,
+    });
+    GoogleAnalytics.pageview(page);
+  }
 };
 
-if (GA_CODE.length) {
-  GoogleAnalytics.initialize(GA_CODE, {
-    testMode: GA_ENV === 'test',
-  });
-  GoogleAnalytics.set({
-    env: GA_ENV,
-    username,
-  });
-}
+/**
+ * helper function to initialize GoogleAnalytics
+ * @param {FlexObject} options tracking options for the page view
+ */
+export const initGoogleAnalytics = (options?: FlexObject) => {
+  const gaCode = getGaCode(options);
+  if (gaCode && gaCode.length) {
+    GoogleAnalytics.initialize(GA_CODE, {
+      testMode: GA_ENV === 'test',
+    });
+    GoogleAnalytics.set({
+      env: GA_ENV,
+      username,
+    });
+  }
+};
 
 /**
  * Higher Order Component (HOC) which handles Google Analytics page view tracking
  * @param {any} WrappedComponent the component to be wrapped by the HOC component
  * @param {FlexObject} options tracking options for the page view
  * @returns HOC rendering the WrappedComponent
  */
-const WithGATracker = (WrappedComponent: any, options: FlexObject = {}) => {
-  const HOC = class extends Component<Props> {
+const WithGATracker = (WrappedComponent: any, options?: FlexObject) => {
+  const gaCode = getGaCode(options);
+  if (!gaCode || !gaCode.length) {
+    return WrappedComponent;
+  }
```

Comment:
This is the actual change which allows us to turn off GA tracking by not providing a GA_CODE.